tests/suites/storage/model_storage.sh

```diff
@@ -0,0 +1,143 @@
+	block_storage_type=""
+	case "${BOOTSTRAP_PROVIDER:-}" in
+	"aws")
+		block_storage_type="ebs"
+		;;
+	"lxd")
+		# Skip block storage for LXD
+		block_storage_type=""
+		;;
+	*)
+		# Default to loop for other providers
+		block_storage_type="loop"
+		;;
+	esac
+
+	# Run filesystem test for all providers
+	test_default_filesystem_storage "${model_name}"
+
+	# Run block storage test for all providers except LXD
+	if [ -n "${block_storage_type}" ]; then
+		test_block_storage "${model_name}" "${block_storage_type}"
+	else
+		echo "Skipping block storage test for ${BOOTSTRAP_PROVIDER:-} provider"
+	fi
+
+	destroy_model "${model_name}"
+}
+
+# Test filesystem storage uses the model's default storage type
+test_default_filesystem_storage() {
+	local model_name=$1
+
+	echo "Testing model config filesystem storage type with tmpfs"
+
+	# Set default filesystem storage type in model config
+	juju model-config -m "${model_name}" storage-default-filesystem-source=tmpfs
+
+	# Deploy application with filesystem storage without specifying storage type
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage --storage single-fs=1G
+
+	# Wait for the application to be active
+	wait_for "dummy-storage" "$(active_condition "dummy-storage" 0)"
+	wait_for "active" "$(workload_status "dummy-storage" 0).current"
+
+	# Verify tmpfs storage details
+	wait_for_storage "attached" '.storage["single-fs/0"]["status"].current'
+	assert_storage "filesystem" "$(kind_name "single-fs" 0)"
+	assert_storage "alive" "$(life_status "single-fs" 0)"
+	assert_storage "dummy-storage/0" "$(unit_attachment "single-fs" 0 0)"
+
+	echo "Filesystem storage type with tmpfs test PASSED"
+	# Clean up
+	juju remove-application -m "${model_name}" --force --no-prompt dummy-storage
+	wait_for "{}" ".applications"
+
+	echo "Testing model config filesystem storage type with rootfs"
+
+	# Set default filesystem storage type in model config
+	juju model-config -m "${model_name}" storage-default-filesystem-source=rootfs
+
+	# Deploy application with filesystem storage without specifying storage type
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage --storage single-fs=1G
+
+	# Wait for the application to be active
+	wait_for "dummy-storage" "$(active_condition "dummy-storage" 0)"
+	wait_for "active" "$(workload_status "dummy-storage" 1).current"
+
+	# Verify rootfs storage details
+	wait_for_storage "attached" '.storage["single-fs/1"]["status"].current'
+	assert_storage "filesystem" "$(kind_name "single-fs" 1)"
+	assert_storage "alive" "$(life_status "single-fs" 1)"
+	assert_storage "dummy-storage/1" "$(unit_attachment "single-fs" 1 0)"
+
+	echo "Filesystem storage type with rootfs test PASSED"
+
+	# Clean up
+	juju remove-application -m "${model_name}" --force --no-prompt dummy-storage
+	wait_for "{}" ".applications"
+
+	echo "All model config filesystem storage tests PASSED"
+}
+
+# Generic test for block storage using the provided storage type
+test_block_storage() {
```

Comment:
As with `test_default_filesystem_storage` this should be called `test_default_block_storage`

Reply:
fixed.